### ISSUE TITLE
osu!droid on iOS

### ISSUE THREAD

**Issue by vargamilan15** [osudroid/osu-droid#431]
Hi! My name is Milán, I’m currently 14 y/o and I live in Hungary. My suggestion is that maybe you guys could bring osu!droid to iOS devices? I know it sounds stupid because of the name, but hear me out. So the new osu!lazer on iOS devices is full of shit, its just the pc version with some drivers installed. From what I know, osu!droid is the best version of osu you can play on mobile devices, and I think you guys can implant it to iPhones and iPads. The easiest method is, that you create an .ipa file, which you can sideload via Altstore. You would need a computer with Win 10/11, an iPhone and a cable to play this amazing game on your phone.

**Comment by Reco1I** [comment]
The game is built in a way is imposible to port to iOS. It has been designed to be used in Android only. That being said, porting it to iOS would require to almost rework the entire game from scratch.

tl;dr: Unfortunately it's not possible.

**Comment by Detusndx** [comment]
It’s probably not as easy as you’re making it out to be, osu!droid uses a version of [AndEngine](https://github.com/nicolasgramlich/AndEngine) which is designed primarily for android devices. The only option to have it running on ios would be to either modify or port it to a different framework to one that supports ios, both of which is a long and tedious process. (OpenGL ES was dropped past iOS 12.0 so you’d have to look into something like MetalAngle or MoltenGL to run under it)
 
It’s definitely not impossible but a lot of work, easier to start a new project at that point, perhaps under an engine that supports both platforms.
